web/packages/pop-miner/src/types.ts

```diff
@@ -0,0 +1,404 @@
+   * call. Errors with this code are likely caused by incorrect usage of this
+   * package.
+   */
+  InvalidValue = 2000,
+}
+
+/**
+ * Represents an error that occurred within the PoP Miner WASM.
+ *
+ * If a Promise returned by an API function is rejected, it will be rejected
+ * with this type.
+ */
+export type Error = {
+  /**
+   * The error code for this error, used to differentiate between error types.
+   */
+  readonly code: ErrorCode;
+
+  /**
+   * The error message.
+   */
+  readonly message: string;
+
+  /**
+   * The Go debug stack for the error.
+   */
+  readonly stack: string;
+
+  /**
+   * The timestamp the error occurred.
+   */
+  readonly timestamp: Date;
+};
+
+/**
+ * Version information for the WASM PoP Miner.
+ *
+ * @see version
+ */
+export type VersionResult = {
+  /**
+   * The version of the WASM PoP Miner.
+   */
+  version: string;
+
+  /**
+   * The SHA-1 hash of the Git commit the WASM binary was built from.
+   */
+  gitCommit: string;
+};
+
+/**
+ * Returns version information for the WASM binary.
+ */
+export declare function version(): Promise<VersionResult>;
+
+/**
+ * @see generateKey
+ */
+export type GenerateKeyArgs = {
+  /**
+   * Determines which network for which the key will be generated.
+   */
+  network: 'testnet3' | 'mainnet';
+};
+
+export type GenerateKeyResult = {
+  /**
+   * The Ethereum address for the generated key.
+   */
+  readonly ethereumAddress: string;
+
+  /**
+   * The network for which the key was generated.
+   */
+  readonly network: 'testnet3' | 'mainnet';
+
+  /**
+   * The generated secpk256k1 private key, encoded as a hexadecimal string.
+   */
+  readonly privateKey: string;
+
+  /**
+   * The secpk256k1 public key for the generated key, in the 33-byte compressed
+   * format, encoded as a hexadecimal string.
+   */
+  readonly publicKey: string;
+
+  /**
+   * The Bitcoin pay-to-pubkey-hash address for the generated key.
+   */
+  readonly publicKeyHash: string;
+};
+
+/**
+ * Generates returns a new secp256k1 private key and its corresponding public
+ * key and addresses.
+ *
+ * @param args Key generation parameters.
+ */
+export declare function generateKey(
+  args: GenerateKeyArgs,
+): Promise<GenerateKeyResult>;
+
+/**
+ * @see startPoPMiner
+ */
+export type MinerStartArgs = {
+  /**
+   * The network to start the PoP Miner on.
+   */
+  network: 'testnet' | 'devnet' | 'mainnet';
+
+  /**
+   * The secpk256k1 private key for the PoP Miner.
+   */
+  privateKey: string;
+
+  /**
+   * The log level for the PoP miner. This controls the verbosity of logs sent
+   * to the console.
+   *
+   * Options are 'trace', 'debug', 'info', 'warn', 'error' and 'critical'.
+   *
+   * @remarks
+   * The logging library used by the PoP miner allows more granular control over
+   * logging levels for individual components. For more information, please
+   * see https://github.com/juju/loggo#func-configureloggers
+   */
+  logLevel?: string | undefined;
```

Comment:
If 

> Options are 'trace', 'debug', 'info', 'warn', 'error' and 'critical'.

What about then explicitly setting those options?

```suggestion
  logLevel?: 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'critical'
```

The `| undefined` could be removed considering `logLevel` is already an optional property... unless we want to allow `logLevel: undefined` (which is still correct, but makes little sense for the caller IMHO)

Reply:
I had considered doing this, however these aren't the only options.
As noted in the documentation above, the underlying logging library used in the PoP miner actually supports a more complex format that allows changing the logging levels for individual components: https://github.com/juju/loggo#func-configureloggers

Would it be worth trying to document this better in the comment, or perhaps improve the type instead of it just being 'string'?

I will remove the `| undefined`, thanks!

Reply:
Their specification to call that function shows it as `string`, so then `string` should be enough then, you're correct.